src/turbomind/layers/sampling_layers/TopKSamplingLayer.cc

```diff
@@ -143,49 +142,57 @@ void TopKSamplingLayer<T>::setup(const size_t batch_size, const size_t beam_widt
     //     temperature [1] or [batch_size] on cpu, optional
     //     repetition_penalty [1] or [batch_size] on cpu, optional
     TM_LOG_DEBUG(__PRETTY_FUNCTION__);
-    BaseSamplingLayer<T>::setup(batch_size, beam_width, runtime_args);
-
-    uint         tmp_top_k     = 0;
-    const Tensor runtime_top_k = runtime_args->isExist("runtime_top_k") ?
-                                     runtime_args->at("runtime_top_k") :
-                                     Tensor(MEMORY_CPU, TYPE_UINT32, {1}, &tmp_top_k);
     const Tensor runtime_top_p = runtime_args->isExist("runtime_top_p") ? runtime_args->at("runtime_top_p") : Tensor();
-    const size_t runtime_top_k_size = runtime_top_k.size();
     const size_t runtime_top_p_size = runtime_top_p.size();
+    const Tensor runtime_top_k = runtime_args->isExist("runtime_top_k") ? runtime_args->at("runtime_top_k") : Tensor();
+    const size_t runtime_top_k_size = runtime_top_k.size();
+    uint         max_top_k          = runtime_top_k_size > 0 ? runtime_top_k.max<uint>() : 0;
+    float        min_top_p          = runtime_top_p_size > 0 ? runtime_top_p.min<float>() : 0.0f;
+    skip_all_                       = false;
+
+    if (max_top_k == 0 && min_top_p != 0.0f) {
```

Comment:
what does `max_top_k==0 && min_top_p==0.0f` indicate?

Reply:
I get it in `setup_topk_runtime_args`

Reply:
What's the relation between `top_k` and `n_logprobs`? Is `n_logprobs` not greater than `top_k?
I just wonder if `skip_all` affects returning `logits` or `logprobs`